Algorithm Codes/Graph/AdjacencyMatrixGraph.py

```diff
@@ -42,13 +46,42 @@ def dijkstra(self, start):
                         heapq.heappush(heap, (dist[i], i))
             
         return (dist, prev)
+    
+    def bellman_ford(self, start):
+        dist = [float('inf') for i in range(self.n)]
+        dist[start] = 0
+        prev = [None for i in range(self.n)]
+
+        for i in range(self.n-1):
+            for j in range(self.n):
+                for k in range(self.n):

```

Comment:
I think the complexity of Original Bellman-Ford algorithm is supposed to be O(V * E) for performing repeated relaxations.

I see the time complexity of this is O(V ^ 3)

I think it would be even better if we could add Negative Cycle Detection on Bellman Ford.
As far as I know, people choose BellmanFord Algorithm when there's a negative cycle in the graph which can't be calculated by Dijkstra.



https://www.geeksforgeeks.org/bellman-ford-algorithm-dp-23/
I think maybe this article might come useful to you!

Reply:
My does return None if a negative cycle exists, are you saying you want me to determine the nodes in the negative cycles, and also the time complexity is O(V^3) because we are using an adjacency matrix to find the edges but I could create a list of the edges at the start of the code in O(V^2) time and just loop through that to have the code have O(EV) runtime.

Reply:
Oh Sorry that I misunderstood the code :crying_cat_face: 
I think you are right regarding the algorithm.
The one I posted is O(V*E) cause it's having explicitly given edges which is not our case.
Resolving this!